File: Data Structures & Algorithms/two-integer-sum/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> complements;
        for (int i =0; i<nums.size(); i++){
            int complement = target - nums[i];
            if (complements.contains(complement)){
                return {complements[complement], i};
    
            }
            complements[nums[i]] = i;
        }
        return {};
    }
};

File: Data Structures & Algorithms/search-insert-position/submission-2.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        l =0
        r = len(nums)-1
        m=0
        i=0
        while l<=r:
            m = (r+l)//2
            if target>nums[m]:
                l = m+1
                i=l
            elif target<nums[m]:
                r = m -1
                i=m
            else:
                i=m
                break
        return i   